packages/js/components/src/sortable/sortable.tsx

```diff
@@ -247,6 +248,14 @@ export const Sortable = ( {
 						items.length
 					),
 				} );
+
+				if (
+					child.props.className &&
+					child.props.className.indexOf( 'not-sortable' ) !== -1
```

Comment:
I'm not sure an HoC approach would be the best here given the issues with wrapping `children` with any components; it could encourage behavior that will cause issues (as I did earlier). Also the individual child that this will apply to will be rotating and changing due to interactions.

For now I switched it to the `notSortableIndexes` prop within the `<Sortable />` component, although this is a little rigid as it only accepts indexes. We could make it more flexible by allowing it to accept child `keys` as well, although this seems like a awkward/rare use case in general.